workspaces/liferay-ticket-workspace/client-extensions/liferay-ticket-custom-element/src/services/listTypeEntries.ts

```diff
@@ -51,9 +44,15 @@ export async function fetchListTypeDefinitions() {
 	const listTypeDefinitions = {} as ListTypeDefinitions;
 
 	for (const listTypeDefinitionERC of listTypeDefinitionERCs) {
-		listTypeDefinitions[listTypeDefinitionERC] = await fetchListTypeEntries(
-			listTypeDefinitionERC
-		);
+		const entries = await fetchListTypeEntries(listTypeDefinitionERC);
+
+		listTypeDefinitions[listTypeDefinitionERC] = {};
+		listTypeDefinitions[listTypeDefinitionERC].array = entries;
+		listTypeDefinitions[listTypeDefinitionERC].map = {};
+
+		entries.forEach((entry: any) => {
+			listTypeDefinitions[listTypeDefinitionERC].map[entry.name] = entry;
+		});
```

Comment:
I would avoid mutations like this because it can lead to unexpected results, it would be better to have it more declarative.